pkg/metrics/metric/counter.go

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package metric
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+func NewCounter(opts CounterOpts) Counter {
+	return &counter{
+		Counter: prometheus.NewCounter(opts.toPrometheus()),
+		metric: metric{
+			enabled: !opts.Disabled,
+			opts:    Opts(opts),
+		},
+	}
+}
+
+type Counter interface {
+	prometheus.Counter
+	WithMetadata
+
+	Get() float64
```

Comment:
Its meant to replace `metrics.GetCounterValue` and `metrics.GetGaugeValue` in future PRs. These are used both in tests and the metricsmap which is a bit of an oddball. 

I never knew `prometheus/testutil` existed, just thought I would replace the above with a nice methods since we are wrapping anyway. I have no personal preference.

Reply:
I don't have a strong preference either. My only concern is that if being part of the interface they might encourage a bit more incorrect usages outside of testing.